### PR TITLE
First implementation of floating screen

### PR DESCRIPTION
**Pull request by RedBrumbler**
Just creating the PR ahead of time, this actually still needs some testing

### CODE REVIEW

**Comment by RedBrumbler** [comment]
Was tested somewhat, seems to work

File: src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                }
+            }
+        }
+
+        // if:
+        // - no grabbing controller, or
+        // - fpfc & controller pressed, or
+        // - fpfc & mouse pressed
+        // early return
+        if (!(_grabbingController && _grabbingController->m_CachedPtr.m_value) ||
```
Comment (raineio) [inline src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:85]: Is this actually required, given we don't have FPFC in the conventional sense?

File: src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        }
+
+        // if:
+        // - no grabbing controller, or
+        // - fpfc & controller pressed, or
+        // - fpfc & mouse pressed
+        // early return
+        if (!(_grabbingController && _grabbingController->m_CachedPtr.m_value) || 
+            (!get_isFpfc() && _grabbingController->get_triggerValue() > 0.9f) || 
+            (get_isFpfc() && UnityEngine::Input::GetMouseButton(0))) return;
```
Comment (raineio) [inline src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:87]: Is this check actually required? We don't have mouse inputs in the conventional sense.

Comment (RedBrumbler) [reply src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:87]: just implemented it cause I didn't wanna have to think about how to not have them lol

Comment (raineio) [reply src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:87]: Fair enough, looks good.

File: src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+namespace BSML {
+    void FloatingScreenMoverPointer::Init(FloatingScreen* floatingScreen, VRUIControls::VRPointer* pointer) {
+        _floatingScreen = floatingScreen;
+        _screenHandle = floatingScreen->handle->get_transform();
+        _realPos = floatingScreen->get_transform()->get_position();
+        _realRot = floatingScreen->get_transform()->get_rotation();
+        _vrPointer = pointer;
+        _fpfc = UnityEngine::Resources::FindObjectsOfTypeAll<GlobalNamespace::FirstPersonFlyingController*>().FirstOrDefault();
```
Comment (raineio) [inline src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:50]: Not required, PC specific code that will do nothing on Quest

File: src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    void FloatingScreenMoverPointer::Init(FloatingScreen* floatingScreen) {
+        auto vrPointer = GetComponent<VRUIControls::VRPointer*>();
+        Init(floatingScreen, vrPointer);
+    }
+
+    bool FloatingScreenMoverPointer::get_isFpfc() {
```
Comment (raineio) [inline src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:58]: Non required method

File: src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp (hunk 1 above)
Comment (raineio) [inline src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:85]: Non-required if check

**Review by raineio — CHANGES_REQUESTED** [review]
These are PC specific checks for FPFC, and will do practically nothing on Quest, as we don't do any funny business in FPFC mode.

File: src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp (hunk 2 above)
Comment (raineio) [reply src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp:87]: continued under, with requested changes

File: src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp (hunk 1 above)
Comment (RedBrumbler) [reply src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp]: I hope you mean the part where it tries using the fpfc & mouse stuff

Comment (raineio) [reply src/BSML/FloatingScreen/FloatingScreenMoverPointer.cpp]: Yep, worded it wrong.

**Review by raineio — APPROVED** [review]
Looks good.

**Pull request merged**